lochness/__init__.py

```diff
@@ -43,6 +43,48 @@
 ])
 
 
+class Subject(object):
+    def __init__(self, active, phoenix_study, phoenix_id, consent, beiwe,
+                 icognition, saliva, xnat, redcap, dropbox,
+                 box, mediaflux, mindlamp, daris, rpms,
+                 general, protected, metadata_file):
+        self.active = active
+        self.study = phoenix_study
+        self.id = phoenix_id
+        self.consent = consent
+        self.beiwe = beiwe
+        self.icognition = icognition
+        self.saliva = saliva
+        self.xnat = xnat
+        self.redcap = redcap
+        self.dropbox = dropbox
+        self.box = box
+        self.mediaflux = mediaflux
+        self.mindlamp = mindlamp
+        self.daris = daris
+        self.rpms = rpms
+        self.general_folder = general
+        self.protected_folder = protected
+        self.metadata_csv = metadata_file
+
+    def asdict(self):
+        '''emulating collection._asdict()'''
+        return {'active': self.active, 'study': self.study,
+                'id': self.id, 'consent': self.consent,
+                'beiwe': self.beiwe, 'icognition': self.icognition,
+                'saliva': self.saliva, 'xnat': self.xnat,
+                'redcap': self.redcap, 'dropbox': self.dropbox,
+                'box': self.box, 'mediaflux': self.mediaflux,
+                'mindlamp': self.mindlamp, 'daris': self.daris,
+                'rpms': self.rpms, 'general_folder': self.general_folder,
+                'protected_folder': self.protected_folder,
+                'metadata_csv': self.metadata_csv}
+        
+    def to_bids(self):
+        self.general_folder = os.path.join(self.general_folder, self.study)
+        self.protected_folder = os.path.join(self.protected_folder, self.study)
```

Comment:
This feels a little dangerous, because `.to_bids()` reads like it should be "change to BIDS mode", and thus running it twice should be the same as running it once.

As a minimal change, setting a class variable `_bids = False`, you could then use it as a flag to prevent doubling up:

```suggestion
        if not self._bids:
            self.general_folder = os.path.join(self.general_folder, self.study)
            self.protected_folder = os.path.join(self.protected_folder, self.study)
            self._bids = True
```

Reply:
Thanks @effigies for the comments. I'll update the code according to your suggestion.